chore(policy): add a CONTRIBUTING.md file (#98)
ecolabdata/ecospheres-front: CONTRIBUTING.md

TL;DR:
- Communication is in [French](https://www.legifrance.gouv.fr/loda/article_lc/LEGIARTI000006421209).
- Code and code documentation are in [English (BR)](https://en.wikipedia.org/wiki/English_as_a_lingua_franca).
- Code documentation is in the [JSDoc](https://jsdoc.app/) format.
- Commits follow [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/).
- Code styling is progamatically enforced and follows [Standard](https://standardjs.com/).

> - Testing is tactical and is done with [Vutest](https://vutest.dev) and [Cypress](https://www.cypress.io/).

Reviewer: Je vois ce que tu veux dire avec "tactical" vu qu'on en a parlé lors du dernier hebdo, mais je suis pas certain que ça soit clair pour un non-initié ?

Author: Let's give it a thought, I'm out of ideas on thato ne.

Reviewer: On peut mettre "Testing is done with..." pour l'instant et voir plus tard.
Quitte à ajouter une section "testing" dédiée plus tard ?

> - Testing is tactical and is done with [Vutest](https://vutest.dev) and [Cypress](https://www.cypress.io/).

Author: ```suggestion
- Testing is done with [Vutest](https://vutest.dev) and [Cypress](https://www.cypress.io/).
```

TL;DR:
- Communication is in [French](https://www.legifrance.gouv.fr/loda/article_lc/LEGIARTI000006421209).

> - Code and documentation are in [English (BR)](https://en.wikipedia.org/wiki/English_as_a_lingua_franca).

Reviewer: Quid de la doc en général (vs les commentaires dans le code) ?
Si on envisage une doc orientée user, le français serait plus pertinent.
Est-ce qu'on se complique la vie en ayant du jsdoc en EN et une doc en FR ?

Author: Je pensais à la doc déjà présente dans le code, je n'avais pas la doc en général en tête ni dans la feuille de route à court terme.

> - Code and documentation are in [English (BR)](https://en.wikipedia.org/wiki/English_as_a_lingua_franca).

Author: ```suggestion
- Code and code documentation are in [English (BR)](https://en.wikipedia.org/wiki/English_as_a_lingua_franca).
```

> - Documentation is in the [JSDoc](https://jsdoc.app/) format.

Author: ```suggestion
- Code documentation is in the [JSDoc](https://jsdoc.app/) format.
```

> - Branches and commits follow [Conventional Commits](https://dev.to/varbsan/a-simplified-convention-for-naming-branches-and-commits-in-git-il4).

Reviewer: Perso pas fan de ce modèle à triple étage, et qui plus est ça pose problème :
```
$ git checkout -b feat/123
Switched to a new branch 'feat/123'
$ git checkout -b feat/123/foo
fatal: cannot lock ref 'refs/heads/feat/123/foo': 'refs/heads/feat/123' exists; cannot create 'refs/heads/feat/123/foo'
```

Proposition : `feat/123-foo`

Diff par rapport au blog cité : 
- 2 étages, pas de pb avec l'exemple ci-dessus : `feat/123` et `feat/123-foo` peuvent co-exister.
- Pas de str "issue-" explicite (on se doute que c'est une référence à un ticket)
- Plus simple si pas de référence : `feat/foo`

Je suis partagé sur le préfixe de type (feat, bug, ...) car c'est parfois pas si évident de faire la distinction, mais je ne suis pas opposé à son utilisation, surtout si on envisage un bénéfice quelconque à ce format (par ex génération de changelog ?)

Author: Je pensais que tu avais proposé conventional commits, autant pour moi. Je suis agnostique.

Author: ```suggestion
- Commits are written to [communicate intent](https://initialcommit.com/blog/git-commit-messages-best-practices).
```

Author: Ça te va cette 2e proposition ?

Reviewer: Conventional commit ne me pose pas de problème. Je connaissais pas mais ça me semble bien.
C'est au niveau du nommage de branches que j'ai qqs soucis.

Reviewer: No messages se sont croisés !
Je propose de distinguer commits et branches. Pour commits, Conventional me semble ok. Pour branches, je n'ai pas de référence pour ce que j'ai proposé, soit on donne le template, soit je peux essayer de chercher un lien.

> - Code styling is imperative and follows [Standard](https://standardjs.com/).

Author: _Imperive_ dans le sens de pas facultatif, pas de _imperative vs functional_.

Reviewer: J'utiliserais "mandatory" dans ce cas.

Author: J'ai mis « programatically enforced », ça te va ?

> - Code styling is imperative and follows [Standard](https://standardjs.com/).

Reviewer: Ça peut être automatisé ?

Author: Oui ! Il y a même déjà une PR : #90

> - Code styling is imperative and follows [Standard](https://standardjs.com/).

Author: ```suggestion
- Code styling is progamatically enforced and follows [Standard](https://standardjs.com/).
```

branch.

Opening a Pull Request means you want that code to be merged. If you want to 
only discuss it, then open a draft and send a link to your draft along with 
your questions through whichever communication channel you prefer.

### Peer reviews

All pull requests must be reviewed by someone else than their original author.

To help reviewers, make sure to add to your PR a **clear text explanation** 
of your changes.

In case of breaking changes, you **must** give details about what features were 
deprecated.

> You must also provide guidelines to help contributors adapt their code to be 
> compatible with the new version of the app.

## Advertising changes

### Version number

We follow the [semantic versioning](http://semver.org/) spec: any change 
impacts the version number, and the version number conveys public interface 
compatibility information **only**.

Examples:

#### Patch bump

- Internal optimization (such as store cache) with no consequence on the stores
  public interface. 

#### Minor bump

- Adding a helper, adding a component, implementing a feature, etc.

#### Major bump

- Renaming or deprecating a component.

> - Breaking changes to a store's public interface.

Reviewer: "store" n'est qu'un exemple => "Breaking changes to a public interface"

> - Breaking changes to a store's public interface.

Author: ```suggestion
- Breaking changes to a public interface.
```

> ### Changelog

Author: ```suggestion
```

> 

Author: ```suggestion
```

> Changelog is generated automatically based on commit and merge messages.

Reviewer: C'est en place ? Si non, ça va être géré comment ?

Author: Non, ça va dépendre je pense de la structure des messages de commit. On peut laisser cela à côté pour le moment.

Reviewer: Si on utilise conventional commit ça semble effectivement facile à automatiser.

Author: Oui, c'est l'idée. Par exemple, sur openfisca, on propose [cette structure](https://openfisca.org/doc/contribute/commit-messages.html), et on regrette ne l'avoir pas fait avant (on continue d'écrire le CHANGELOG à la main).

Author: Je l'ai supprimé pour l'instant.

> 

Author: ```suggestion
```

## Error messages

> Ecosphères provides both an app for end users, and a template for developers.

Reviewer: Je ne vois pas ce que ça fait là

> Ecosphères provides both an app for end users, and a template for developers.

Author: ```suggestion
```

> 

Author: ```suggestion
```
